indico/modules/rb/client/js/__tests__/util.test.js

```diff
@@ -0,0 +1,153 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2023 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import moment from 'moment';
+
+import {renderRecurrenceWeekdays} from '../util';
+
+describe('renderRecurrenceWeekdays', () => {
+  it('should return null if weekdays array is empty', () => {
+    const weekdays = [];
+    expect(renderRecurrenceWeekdays(weekdays)).toBeNull();
+  });
+
+  it('should return null if weekdays array is null', () => {
+    const weekdays = null;
+    expect(renderRecurrenceWeekdays(weekdays)).toBeNull();
+  });
+
+  it('should handle a single unknown/bad weekday', () => {
+    const weekdays = ['xyz'];
+    expect(renderRecurrenceWeekdays(weekdays)).toBeNull();
+  });
+
+  it('should handle multiple unknown/bad weekdays', () => {
+    const weekdays = ['xyz', 'abc', 'def'];
+    expect(renderRecurrenceWeekdays(weekdays)).toBeNull();
+  });
+
+  it('correctly formats a single weekday', () => {
+    const formatted = renderRecurrenceWeekdays(['mon']);
+    expect(formatted).toBe('Monday');
+  });
+
+  it('correctly formats two weekdays', () => {
+    const formatted = renderRecurrenceWeekdays(['mon', 'tue']);
+    expect(formatted).toBe('Monday and Tuesday');
+  });
+
+  it('correctly formats multiple weekdays', () => {
+    const formatted = renderRecurrenceWeekdays(['mon', 'tue', 'wed']);
+    expect(formatted).toBe('Monday, Tuesday, and Wednesday');
+  });
+
+  it('correctly formats weekdays in non-sequential order', () => {
+    const formatted = renderRecurrenceWeekdays(['wed', 'mon', 'fri']);
+    expect(formatted).toBe('Monday, Wednesday, and Friday');
+  });
+
+  it('should handle all weekdays', () => {
+    const weekdays = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'];
+    const expected = 'Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, and Sunday';
+    expect(renderRecurrenceWeekdays(weekdays)).toBe(expected);
+  });
+
+  it('should handle all weekdays in non-sequential order', () => {
+    const weekdays = ['wed', 'mon', 'fri', 'tue', 'sun', 'thu', 'sat'];
+    const expected = 'Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, and Sunday';
+    expect(renderRecurrenceWeekdays(weekdays)).toBe(expected);
+  });
+});
+
+describe('renderLocalizedRecurrenceWeekdays', () => {
+  function mockLocale(locale) {
+    // Set the locale to the one we want to test
+    moment.locale(locale);
+  }
+
+  function restoreLocale() {
+    // Reset the locale to the default one
+    return () => moment.locale();
+  }
+
+  afterEach(() => {
+    restoreLocale();
+  });
+
+  const unorderedWeekdays = ['wed', 'mon', 'fri', 'tue', 'sun', 'thu', 'sat'];
+
+  it('should localize all weekdays in non-sequential order in German (DE)', () => {
```

Comment:
There's lots of repetition here, what about using `test.each`?

```js
  const unorderedWeekdays = ['wed', 'mon', 'fri', 'tue', 'sun', 'thu', 'sat'];
  const cases = [
    ['de', 'Montag, Dienstag, Mittwoch, Donnerstag, Freitag, Samstag und Sonntag'],
    ['fr', 'lundi, mardi, mercredi, jeudi, vendredi, samedi et dimanche'],
  ];

  test.each(cases)(
    'should localize all weekdays in non-sequential order in %p locale',
    (locale, expected) => {
      const oldLocale = moment.locale();
      moment.locale(locale);
      const formatted = renderRecurrenceWeekdays(unorderedWeekdays);
      expect(formatted).toBe(expected);
      moment.locale(oldLocale);
    }
  );
```

Reply:
Yep - this is much nicer :)